modules/odc-uninstalling-web-terminal.adoc

```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// web_console/odc-about-web-terminal.adoc
+
+[id="odc-uninstalling-web-terminal_{context}"]
+= Uninstalling the web terminal
+
+To uninstall the web terminal, you must first delete the components and Custom Resources (CRs) that were added by default when you installed the Operator, and then uninstall the *Web Terminal* Operator.
+
+Uninstalling only the *Web Terminal* Operator will not remove any of its custom resource definitions or managed resources created by default when the Operator is installed. These components must be manually uninstalled for security purposes. It also allows you to save cluster resources as terminals cannot be idled when the Operator is uninstalled.
+
+.Prerequisites
+* Access to an OpenShift Container Platform cluster using an account with `cluster-admin` permissions.
+
+== Deleting the web terminal components and custom resources
+
+Use the CLI to delete the Custom Resources (CRs) created by default during installation of the  *Web Terminal* Operator.
+
+.Procedure
+. Ensure that all `DevWorkspace` CRs are removed along with their related Kubernetes objects, such as, deployments. It is crucial that this is done first, otherwise finalizers make it difficult to fully uninstall the operator.
++
+----
+oc delete devworkspaces.workspace.devfile.io --all-namespaces --all --wait
+oc delete workspaceroutings.controller.devfile.io --all-namespaces --all --wait
+oc delete components.controller.devfile.io --all-namespaces --all --wait
+----
++
+. Remove the custom resource definitions:
++
+----
+oc delete customresourcedefinitions.apiextensions.k8s.io workspaceroutings.controller.devfile.io
+oc delete customresourcedefinitions.apiextensions.k8s.io components.controller.devfile.io
+oc delete customresourcedefinitions.apiextensions.k8s.io devworkspaces.workspace.devfile.io
+----
++
+. Remove the `DevWorkspace-Webhook-Server` deployment:
++
+----
+oc delete deployment/devworkspace-webhook-server -n openshift-operators
+----
++
+[NOTE]
+====
+When you run this and the following steps, you cannot use the `oc exec` commands to run commands in a container. Once you remove the webhooks you will be able to use the `oc exec` commands again.
+====
++
+. Remove any lingering service, secrets, and configmaps:
++
+----
+oc delete all --selector app.kubernetes.io/part-of=devworkspace-operator,app.kubernetes.io/name=devworkspace-webhook-server
+oc delete serviceaccounts devworkspace-webhook-server -n openshift-operators
+oc delete configmap devworkspace-controller -n openshift-operators
+oc delete clusterrole devworkspace-webhook-server
+oc delete clusterrolebinding devworkspace-webhook-server
+----
++
+. Remove mutating or validating webhook configurations:
++
+----
+oc delete mutatingwebhookconfigurations controller.devfile.io
+oc delete validatingwebhookconfigurations controller.devfile.io
+----
+
+== Uninstalling the Operator using the web console
```

Comment:
I'm not sure if we can move this out from the uninstallation procedure. It may lead to unexpected behavior when operator is restarted and recreated some of the previously removed.

If we set mandatory order as
- Uninstalling the Operator using the web console
- Deleting the web terminal components and custom resources

then it should work.

@JPinkney WDYT?

Reply:
Sorry for the confusion, the Operator must be removed after all CR and CRDs are removed

if we remove CRs and webhook server - there is a chance that there will be recreated before we remove operator.

If we remove operator first, there is a possibility that CRs will hang in terminating state, cause finalizers won't be processed.

Reply:
I was going to comment that but you beat me to it :smile:

Reply:
@JPinkney @sleshchenko do I have it correct then? This section needs no changes.
We recommend end users to first Delete the web terminal components and custom resources and then the Uninstall the Operator using the web console. Can you confirm, please?

Reply:
Yes, that flow is correct